core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -179,6 +179,7 @@ package object config {
 
   private[spark] val EXECUTOR_CORES = ConfigBuilder(SparkLauncher.EXECUTOR_CORES)
     .intConf
+    .checkValue(_ > 0, "Each executor must contain at least 1 cpu core.")
```

Comment:
Is it a breaking change?

Reply:
this is probably not needed here as we already check it in SparkSubmitArguments:

Exception in thread "main" org.apache.spark.SparkException: Executor cores must be a positive number
	at org.apache.spark.deploy.SparkSubmitArguments.error(SparkSubmitArguments.scala:658)
	at org.apache.spark.deploy.SparkSubmitArguments.validateSubmitArguments(SparkSubmitArguments.scala:284)

Reply:
removed